### PR TITLE
Make fewer test executables

### PR DESCRIPTION
**Pull request by dneto0**
Try to reduce the amount of disk space used by especially by debug builds,
which may be contributing to AppVeyor failures.

Collapses tests in categories:
- validator
- loop optimizations
- dominator analysis
- linker

Contributes to #1615

### CODE REVIEW

**Comment by dneto0** [comment]
When I run "ctest -j 20" I still get consistent total test times of roughly 10s.
(That's part of why I left validator limits test on its own.  It's a tall pole.)

**Comment by alan-baker** [comment]
Basically the same as #1616?

**Comment by dj2** [comment]
This is basically a small version of https://github.com/KhronosGroup/SPIRV-Tools/pull/1614.

This version maybe faster for the -j20 variant as I have 5 suites and this will have a bunch. The longest suite in my version is 30sec. I don't know if ctest -j20 will run tests in parallel or just suites?

**Comment by dneto0** [comment]
>  I don't know if ctest -j20 will run tests in parallel or just suites? 

It runs suites in parallel.  That is, runs test executables in parallel.
(And yah, I didn't see your PR before starting on mine.  :-)

**Comment by dj2** [comment]
I'm fine with the big or small version. The big version means you only have to remember on suite name to run, the small version is faster when run parallel. I'm curious what causes the slow down for the big version I can see it pausing every now and then so is it just a few really slow tests or the number of tests that causes the 30sec runtime?

File: test/val/CMakeLists.txt [hunk 1]
```diff
@@ -18,177 +18,54 @@ set(VAL_TEST_COMMON_SRCS
...
-
-add_spvtools_unittest(TARGET val_primitives
-        SRCS val_primitives_test.cpp
-       ${VAL_TEST_COMMON_SRCS}
-  LIBS ${SPIRV_TOOLS}
-)
-
-add_spvtools_unittest(TARGET val_ext_inst
-	SRCS val_ext_inst_test.cpp
+add_spvtools_unittest(TARGET val_abcde
```
Comment (dj2) [inline test/val/CMakeLists.txt]: heh, these are terrible test names .... Is there some logical way to break this apart, like numerics, something, something?

Comment (dneto0) [reply test/val/CMakeLists.txt]: Lol.

The abcde says the test covers  .cpp source files starting from a through e.  I had hoped it was an obvious mapping.

**Comment by dneto0** [comment]
> I can see it pausing every now and then so is it just a few really slow tests or the number of tests that causes the 30sec runtime?

Most tests are super fast.  A small handful of tests are slow because they test limits on things like structured nesting depth and similar combinatorial explosions.

I think there is a happy medium somewhere.   Anyway, it will be good data to let both MRs run their course.

**Comment by dneto0** [comment]
So AppVeyor passed: Yay
Clang-format check failed:  Boo.

Rebased, and fixed formatting.

**Comment by dj2** [comment]
Test names make me sad but lgtm regardless.

**Pull request merged**